sp/src/game/server/episodic/npc_advisor.cpp

```diff
@@ -1407,13 +1417,18 @@ bool CNPC_Advisor::IsShieldOn( void )
 
 void CNPC_Advisor::ApplyShieldedEffects(  )
 {
-	if ( IsShieldOn() )
+	// Modulus with the skin allows multiple sets of shielded / unshielded skins
+	if ( IsShieldOn() && GetSkin() % 2 == ADVISOR_SKIN_NOSHIELD )
 	{
+		EmitSound( "NPC_Advisor.shieldup" );
 		SetBloodColor( DONT_BLEED );
+		SetSkin( GetSkin() + 1 );
 	}
-	else
+	else if ( !IsShieldOn() && GetSkin() % 2 == ADVISOR_SKIN_SHIELD)
 	{
+		EmitSound( "NPC_Advisor.shielddown" );
 		SetBloodColor( BLOOD_COLOR_GREEN );
+		SetSkin( GetSkin() - 1 );
```

Comment:
`GetSkin()` and `SetSkin()` were intended for VScript and code is normally supposed to access `m_nSkin` directly. Not a huge deal, but it's worth mentioning.

```suggestion
	// Modulus with the skin allows multiple sets of shielded / unshielded skins
	if ( IsShieldOn() && m_nSkin % 2 == ADVISOR_SKIN_NOSHIELD )
	{
		EmitSound( "NPC_Advisor.shieldup" );
		SetBloodColor( DONT_BLEED );
		m_nSkin += 1;
	}
	else if ( !IsShieldOn() && m_nSkin % 2 == ADVISOR_SKIN_SHIELD)
	{
		EmitSound( "NPC_Advisor.shielddown" );
		SetBloodColor( BLOOD_COLOR_GREEN );
		m_nSkin -= 1;
	}
```